blog/_posts/2017-02-23-scalafix-v0.3.md

```diff
@@ -0,0 +1,141 @@
+
+To introduce new imports, it's possible to pass in `additionalImports`
+
+```scala
+Replace(Symbol("_root_.cats.data.XorT."), q"EitherT",
+        additionalImports = List(importer"cats.data.EitherT")),
+```
+
+Imports can be removed with the `RemoveGlobalImport` patch
+
+```scala
+RemoveGlobalImport(importer"cats.data.Xor")
+```
+
+Nothing happens if the import does not appear in the source file.
+Likewise, it's OK to add the same import twice, scalafix will de-duplicate it.
+
+The full `Xor` to `Either` scalafix rewrite can be found [here][xor2either] and its accompanying test
+suite [here][xor2either-test].
+
+## Try it out!
+You can use scalafix both as a library and a tool.
+
+The recommended way to use scalafix as a library is with the [`sbt-scalahost`][sbt-scalahost] plugin.
+By using scalafix as a library, you have full control of how, where and when to run rewrites.
+
+The recommended way to use scalafix as tool is the [`sbt-scalafix`][sbt-scalafix] plugin.
+It's possible define custom tree patches in the [`.scalafix.conf` configuration file][config-patches].
+
+The long-term mission of scalafix is to help automate the migration of deprecated Scala 2.x features to Dotty.
+However, as I have hopefully demonstrated in this post, scalafix can be used for more than just migrating between Scala versions.
+Scalafix can also be used for ad-hoc library and application migrations.
+
+I am excited to see what applications the community can build with scala.meta and scalafix.
+Some promising ideas that have floated around include
+
+- parse and run rewrites from `@deprecated` warning messages. This would enable library authors
+  to provide exectuable migration guides.
+- use scalafix to cross-build against multiple similar APIs. For example, a
+  library author could write against scalaz and rewrite to cats.
+- build a code search web-interface with "jump to definition" functionality
+  powered by the scala.meta's semantic database.
+- replace usage of the infamous `any2StringAdd` in favor of string interpolators or explicit `.toString`.
+- replace usages of `scala.Seq`, which can be mutable, in favor of `scala.collection.immutable.Seq`.
+
+If this sounds exciting to you, join us!
+I am happy to answer any question in the [scalafix gitter][gitter] channel.
+
+[xor2either-test]: https://github.com/scalacenter/scalafix/blob/f61136fad79afcdbb03528ce78c7928afc6eafd6/scalafix-nsc/src/test/resources/syntactic/Xor2Either.source
+[xor2either]: https://github.com/scalacenter/scalafix/blob/f61136fad79afcdbb03528ce78c7928afc6eafd6/core/src/main/scala/scalafix/rewrite/Xor2Either.scala
+[xor-circe]: https://github.com/circe/circe/pull/343/files
+[cats]: http://github.com/typelevel/cats
+[ghpages]: http://github.com/scalacenter/scalafix
+[prevpost]: http://scala-lang.org/blog/2016/10/24/scalafix.html
```

Comment:
Good catch! Yes, yes old post contains lots of useful background on scala.meta. Just gotta find a way to link it :)